templates/userdata.sh.tpl

```diff
@@ -3,35 +3,14 @@
 # Allow user supplied pre userdata code
 ${pre_userdata}
 
-# Certificate Authority config
-CA_CERTIFICATE_DIRECTORY=/etc/kubernetes/pki
-CA_CERTIFICATE_FILE_PATH=$CA_CERTIFICATE_DIRECTORY/ca.crt
-mkdir -p $CA_CERTIFICATE_DIRECTORY
-echo "${cluster_auth_base64}" | base64 -d >$CA_CERTIFICATE_FILE_PATH
+KUBELET_EXTRA_ARGS=${kubelet_extra_args}
 
 # Set kubelet --node-labels if kubelet_node_labels were set
 KUBELET_NODE_LABELS=${kubelet_node_labels}
-if [[ $KUBELET_NODE_LABELS != "" ]]; then sed -i '/INTERNAL_IP/a \ \ --node-labels='"$KUBELET_NODE_LABELS"'\ \\' /etc/systemd/system/kubelet.service; fi
+if [[ $KUBELET_NODE_LABELS != "" ]]; then KUBELET_EXTRA_ARGS="$KUBELET_EXTRA_ARGS --node-labels=$KUBELET_NODE_LABELS"; fi
```

Comment:
This is nice approach but since you have already listed this as a breaking change, can we just remove `KUBELET_NODE_LABELS` from here and from `locals.tf`? I know it won't be backwards compatible but it will be cleaner, right?

Reply:
done